### ISSUE TITLE
Rooms and Portal system crashes after freeing a room and re-converting room graph

### ISSUE THREAD

**Issue by AhNesh** [godotengine/godot#63375]
### Godot version

3.5-rc7

### System information

Windows 10

### Issue description

I am making a system that seamlessly loads and unloads parts of a level as needed. It flows naturally that each level part is a room or roomgroup.

The game sometimes crashes if I am looking towards a level part that has just been unloaded. I get this error:
```
Error: FATAL: Index p_index = 4 is out of bounds (count = 2).
    at: operator[] (./core/local_vector.h:175)
```

I have figured out that it is `VisualServerScene.Camera.previous_room_id_hint` that is being used to `PortalRendered.get_room()`, and that after clearing and re-converting the rooms that id is no longer valid.

`VisualServerScene.Camera.previous_room_id_hint` should probably be invalidated during `PortalRenderer.rooms_and_portals_clear()`, or it's validity should be checked before using it.

### Steps to reproduce

- Download the MRP and run the main scene.
- Use arrow keys to walk into the yellow room. This causes red, blue, and green rooms to be unloaded, and room graph to be rebuilt.
- Return to the middle room. Red, blue, and green rooms get loaded back in and room graph is rebuilt.
- Walk into any of the rooms while looking towards the yellow room.
- The game crashes just as you enter. The error is only printed to console, not the editor Output window.

Also note that room conversion completed before the crash, as seen in the log.

### Minimal reproduction project

[PortalSystemTest.zip](https://github.com/godotengine/godot/files/9175213/PortalSystemTest.zip)

**Comment by lawnjelly** [comment]
Ah yes well spotted, your assessment sounds likely to be correct, I'll take a look. :+1:

**Comment by lawnjelly** [comment]
I have a tentative fix in the PR, but I haven't yet confirmed as I haven't managed to reproduce the bug at runtime yet (I have no doubt it happens though). I'll just do some more looking through before moving it out of draft.

**Comment by akien-mga** [comment]
Fixed by #63387.
